sw/device/sca/lib/sca.c

```diff
@@ -129,29 +143,39 @@ void handler_irq_timer(void) {
       dif_rv_timer_irq_clear(&timer, kRvTimerHart, kRvTimerComparator));
 }
 
-void sca_init(void) {
-  pinmux_init();
-  sca_init_uart();
-  sca_init_gpio();
-  sca_init_timer();
-}
+/**
+ * Disables the given peripherals to reduce noise during SCA.
+ *
+ * Care must be taken when disabling the entropy complex if a peripheral that
+ * depends on it will be used in SCA. E.g., We can disable the entropy complex
+ * when analyzing AES only because AES features a parameter to skip PRNG
+ * reseeding for SCA experiments. Without this parameter, AES would simply get
+ * stalled with a disabled entropy complex.
+ *
+ * @param disable Set of peripherals to disable.
+ */
+void sca_disable_peripherals(sca_peripherals_t disable) {
+  if (disable & kScaPeripheralEdn) {
+    // Disable/stopping functionality not yet provided by EDN DIFs.
+    mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR),
```

Comment:
I would implement these functions directly in the difs as is. Whenever a peripheral needs to be updated, it takes time to find and update all the places that don't consume DIFs.

Reply:
Added a note about `dif_end_stop()` in #5465 and created #7837 for `dif_csrng_stop()`.